frontend/src/component/environments/CreateEnvironmentButton/CreateEnvironmentButton.tsx

```diff
@@ -3,18 +3,29 @@ import Add from '@mui/icons-material/Add';
 import { ADMIN } from 'component/providers/AccessProvider/permissions';
 import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
 import { useNavigate } from 'react-router-dom';
+import { useEnvironments } from 'hooks/api/getters/useEnvironments/useEnvironments';
 
 export const CreateEnvironmentButton = () => {
     const { uiConfig } = useUiConfig();
+    const { environments } = useEnvironments();
+    const environmentLimit = uiConfig.resourceLimits.environments;
     const navigate = useNavigate();
 
+    const atEnvironmentLimit = environments.length >= environmentLimit;
```

Comment:
I used limitReached convention in other places. I like both atLimit and limitReached as long as we agree on consistent naming.